File: win-sender/src/WinSender/Signaling/AuthPayloads.cs
using System.Text.Json.Serialization;

namespace WinSender.Signaling;

public record AuthInfo(
    [property: JsonPropertyName("type")] string Type,
    [property: JsonPropertyName("value")] string Value);

public record ConnectRequestPayload(
    [property: JsonPropertyName("deviceId")] string DeviceId,
    [property: JsonPropertyName("deviceName")] string DeviceName,
    [property: JsonPropertyName("auth")] AuthInfo Auth);

public record ConnectAcceptPayload(
    [property: JsonPropertyName("token")] string Token,
    [property: JsonPropertyName("deviceId")] string DeviceId,
    [property: JsonPropertyName("deviceName")] string DeviceName);

public record ConnectRejectPayload(
    [property: JsonPropertyName("reason")] string Reason);

File: atv-receiver/app/src/main/java/com/atvcast/receiver/signaling/AuthPayloads.kt
package com.atvcast.receiver.signaling

data class AuthInfo(val type: String, val value: String)

data class ConnectRequestPayload(
    val deviceId: String,
    val deviceName: String,
    val auth: AuthInfo
)

data class ConnectAcceptPayload(
    val token: String,
    val deviceId: String,
    val deviceName: String
)

data class ConnectRejectPayload(val reason: String)
